chain/beacon/chainstore.go

```diff
@@ -319,6 +377,8 @@ func (c *chainStore) AppendedBeaconNoSync() chan *chain.Beacon {
 }
 
 type partialInfo struct {
+	spanContext oteltrace.SpanContext
```

Comment:
TL;DR; Yes, it makes tracing more precise.

Long answer:
We need this here because the messages are sent on channels and used in a separate goroutine.

If we'd pass the context/span of that goroutine, then the tracing data won't show the precise moment of the events that process the messages from those channels.